cmd/internal/register/maps.go

```diff
@@ -152,18 +201,29 @@ func Maps(a *atlas.Atlas, maps []config.Map, providers map[string]provider.Tiler
 				maxZoom = uint(*l.MaxZoom)
 			}
 
-			// add our layer to our layers slice
-			newMap.Layers = append(newMap.Layers, atlas.Layer{
-				Name:              string(l.Name),
-				ProviderLayerName: providerLayer[1],
-				MinZoom:           minZoom,
-				MaxZoom:           maxZoom,
-				Provider:          provider,
-				DefaultTags:       defaultTags,
-				GeomType:          layerGeomType,
-				DontSimplify:      bool(l.DontSimplify),
-				DontClip:      	   bool(l.DontClip),
-			})
+			// for each provider layer, get map name
+			// this is a loop to capture auto provider layers with multiple layers that match regex
+			for _, name := range provLayers {
+				var lname string
+				if auto {
+					lname = name
+				} else {
+					lname = string(l.Name)
+				}
+
```

Comment:
Why use `l.Name` and not just the name, value.

Reply:
Because there are different conditions based on whether it is auto-generated or not. For a defined layer, Name and ProviderLayerName will be different. But for an auto-generated name, they should be the same (because otherwise the ProviderLayerName would be in regex).